src/components/TodoContainer.js

```diff
@@ -1,140 +1,84 @@
-
-const AllButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #39393a;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-`;
-
-const ActiveButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #ffe74c;
-  color: #333;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-  margin: 0 10px;
-`;
-
-const CompletedButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #666;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-`;
-
-const TodoList = styled.div`
-  margin-top: 10px;
-`;
-
-const ClearTodo = styled.button`
-  font-family: "ubuntu";
-  width: 120px;
-  background: #c8553d;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-  margin: 10px;
-`;
+import {
+  TodoWrapper,
+  Title,
+  CreateTodo,
+  TodoInput,
+  AllButton,
+  ActiveButton,
+  CompletedButton,
+  TodoList,
+  ClearTodo,
+} from "./styles/TodoContainer.style";
 
 export default function TodoContainer() {
   const id = useRef(1);
   const [todos, setTodos] = useState([]);
   const [value, setValue] = useState("");
   const [filter, setFilter] = useState("all");
 
-  const handleAddTodo = useCallback(() => {
-    if (!value) return alert("wanna type something?");
-    setTodos((todo) => [{ id: id.current, content: value }, ...todos]);
-    setValue("");
-    id.current++;
-  }, [todos, value]);
+  useEffect(() => {
+    if (localStorage.getItem("savedLocalTodos")) {
+      const savedLocalTodos = JSON.parse(
+        localStorage.getItem("savedLocalTodos")
+      );
+      setTodos(savedLocalTodos);
+    }
+  }, []);
+
+  const handleAddTodo = useCallback(
+    (e) => {
+      if (!value || value.trim() === "") return;
+      if (e.key === "Enter") {
+        const todo = { id: id.current, content: value, isDone: false };
+        localStorage.setItem(
+          "savedLocalTodos",
+          JSON.stringify([...todos, todo])
+        );
+        setTodos([...todos, todo]);
+        setValue("");
+        id.current++;
+      }
+    },
+    [todos, value]
+  );
 
   const handleInputChange = useCallback((e) => {
     setValue(e.target.value);
   }, []);
 
   const handleDeleteTodo = useCallback(
     (id) => {
-      setTodos(todos.filter((todo) => todo.id !== id));
+      const leftTodos = todos.filter((todo) => todo.id !== id);
+      localStorage.setItem("savedLocalTodos", JSON.stringify(leftTodos));
+      setTodos(leftTodos);
     },
     [todos]
   );
 
   const handleTodoIsDone = useCallback(
     (id) => {
-      setTodos(
-        todos.map((todo) => {
-          if (todo.id !== id) return todo;
-          return {
-            ...todo,
-            isDone: !todo.isDone,
-          };
-        })
+      const todosCompleteStatus = todos.map((todo) => {
```

Comment:
where is this `todos` from ?

Reply:
> where is this `todos` from ?

it's from line 17 `const [todos, setTodos] = useState([]);`